bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,377 @@
+        #     'EndpintUrl': {
+        #         'Description': 'URL for endpoint',
+        #         'Value': 'Fn::Sub: "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod"'
+        #     }
+        # },
+    }
+    for api_name in api_names:
+        sam_config['Resources'][api_name] = generate_function_resource(
+            deployment_name,
+            api_name,
+            s3_bucket_name,
+            py_runtime,
+            memory_size=memory_size,
+            timeout=timeout,
+        )
+
+    yaml.dump(sam_config, Path(template_file_path))
+
+    # We add Outputs section separately, because the value should not have 'around !Sub
+
+    with open(os.path.join(project_dir, 'template.yaml'), 'a') as f:
+        f.write(
+            """\
+Outputs:
+    EndpointUrl:
+        Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod"
+        Description: URL for endpoint
+"""
+        )
+    return template_file_path
+
+
+class AwsLambdaDeploymentOperator(DeploymentOperatorBase):
+    def apply(self, deployment_pb, yatai_service, prev_deployment):
+        try:
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+            deployment_spec = deployment_pb.spec
+
+            bento_pb = yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    'BentoML currently not support {} repository'.format(
+                        bento_pb.bento.uri.type
+                    )
+                )
+
+            return self._apply(
+                deployment_pb, bento_pb, yatai_service, bento_pb.bento.uri.uri
+            )
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def _apply(self, deployment_pb, bento_pb, yatai_service, bento_path):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._apply(deployment_pb, bento_pb, yatai_service, local_path)
+
+        try:
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+            bento_service_metadata = bento_pb.bento.bento_service_metadata
+
+            # TODO
+            python_runtime = 'python3.7'
+            if version.parse(bento_service_metadata.env.python_version) < version.parse(
+                '3.0.0'
+            ):
+                python_runtime = 'python2.7'
+
+            api_names = (
+                [aws_config.api_name]
+                if aws_config.api_name
+                else [api.name for api in bento_service_metadata.apis]
```

Comment:
If aws_config.api_name is None and len(bento_service_metadata.apis) > 1, let's add a log warning:
logger.warning("When no api_name specified for lambda deployment, BentoML will create one lambda function for every API in BentoService by default. APIs found in target BentoService bundle are {}".format(','.join(list(map(lambda x:x.name, bento_service_metadata.apis))))